src/Sylius/Component/Channel/Context/RequestBased/ChannelContext.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Channel\Context\RequestBased;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class ChannelContext implements ChannelContextInterface
+{
+    /**
+     * @var RequestResolverInterface
+     */
+    private $requestResolver;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @param RequestResolverInterface $requestResolver
+     * @param RequestStack $requestStack
+     */
+    public function __construct(RequestResolverInterface $requestResolver, RequestStack $requestStack)
+    {
+        $this->requestResolver = $requestResolver;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChannel()
+    {
+        $channel = $this->requestResolver->getChannel($this->getMasterRequest());
+
+        $this->assertChannelWasFound($channel);
+
+        return $channel;
+    }
+
+    /**
+     * @return Request
+     */
+    private function getMasterRequest()
+    {
+        $masterRequest = $this->requestStack->getMasterRequest();
+        if (null === $masterRequest) {
```

Comment:
It's a common practice not to put a blank line between those if one line is assigning the value to A and the second one is `if` doing some comparison using A.